src/base/charstr.rs

```diff
@@ -56,6 +56,9 @@ use std::vec::Vec;
 pub struct CharStr<Octs: ?Sized>(Octs);
 
 impl<Octs: ?Sized> CharStr<Octs> {
+    /// Character strings have a maximum length of 255 octets.
+    pub const MAX_LENGTH: usize = 255;
```

Comment:
If you define the constant for a concrete type, for instance, `CharStr<()>`, you avoid having to use the turbofish in some cases and can always refer to it as `CharStr::MAX_LENGTH` – though not `Self::MAX_LENGTH` anymore, but I think that’s fine as these names should be fairly stable.

(This goes for all the other generic types as well.)

Reply:
ah nice yeah I was wondering how to do that